stompServer.js

```diff
@@ -351,18 +375,130 @@ var StompServer = function (config) {
     return frame;
   };
 
-  function parseRequest(socket, data) {
+  //</editor-fold>
+
+
+  //<editor-fold defaultstate="collapsed" desc="Send">
+
+  /**
+   * Heart-beat: Turn On for given socket
+   *
+   * @param {WebSocket} socket Destination WebSocket
+   * @param {number} interval Heart-beat interval
+   * @param {boolean} serverSide If true then server is responsible for sending pings
+   * */
+  this.heartbeatOn = function (socket, interval, serverSide) {
+    var self = this;
+
+    if (serverSide) {
+      // Server takes responsibility for sending pings
+      // Client should close connection on timeout
+      socket.heartbeatClock = setInterval(function() {
+        if(socket.readyState == 1) {
+          self.conf.debug('PING');
+          socket.send(Bytes.LF);
+        }
+      }, interval);
+
+    } else {
+      // Client takes responsibility for sending pings
+      // Server should close connection on timeout
+      socket.heartbeatTime = Date.now();
+      socket.heartbeatClock = setInterval(function() {
+        var diff = Date.now() - socket.heartbeatTime;
+        if (diff > interval) {
+          self.conf.debug('HEALTH CHECK failed! Closing', diff, interval);
+          socket.close();
+        } else {
+          self.conf.debug('HEALTH CHECK ok!', diff, interval);
+        }
+      }, interval * 2 + 1); // give it a nice margin of error
```

Comment:
Are you sure this will work properly? I think the intervals in line 415 and 409 should be switched.
( Checking in _interval_ intervals, but comparing to the value with margin.

Reply:
Hmm I took this solution from client side StompJS library, but you got a point. I have drawn timeline on the paper one more time and yes - this may be a bug. I will fix it.